src/gui/attributetable/qgsattributetablemodel.cpp

```diff
@@ -494,6 +495,19 @@ QVariant QgsAttributeTableModel::headerData( int section, Qt::Orientation orient
       return tr( "feature id" );
     }
   }
+  else if ( role == Qt::DecorationRole && orientation == Qt::Vertical )
+  {
+    QgsRenderContext ctx;
+    layer()->rendererV2()->startRender( ctx, layer()->pendingFields() );
+    QgsFeatureId id = mRowIdMap[section];
+    QgsFeature feature;
+    mLayerCache->featureAtId( id, feature );
+    QgsSymbolV2List symbols = layer()->rendererV2()->symbolsForFeature( feature );
+    QgsSymbolV2* symbol = symbols.first();
```

Comment:
Thanks

On Thu, 9 Jul 2015 6:23 pm Nyall Dawson <notifications@github.com> wrote:

> In src/gui/attributetable/qgsattributetablemodel.cpp
> <https://github.com/qgis/QGIS/pull/2203#discussion_r34232999>:
>
> > @@ -494,6 +495,19 @@ QVariant QgsAttributeTableModel::headerData( int section, Qt::Orientation orient
> >        return tr( "feature id" );
> >      }
> >    }
> > +  else if ( role == Qt::DecorationRole && orientation == Qt::Vertical )
> > +  {
> > +    QgsRenderContext ctx;
> > +    layer()->rendererV2()->startRender( ctx, layer()->pendingFields() );
> > +    QgsFeatureId id = mRowIdMap[section];
> > +    QgsFeature feature;
> > +    mLayerCache->featureAtId( id, feature );
> > +    QgsSymbolV2List symbols = layer()->rendererV2()->symbolsForFeature( feature );
> > +    QgsSymbolV2* symbol = symbols.first();
>
> needs a check here that symbols isn't empty - otherwise crashes if the
> layer is using something like the heatmap renderer.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/qgis/QGIS/pull/2203/files#r34232999>.
>